src/buildblock/ML_norm.cxx

```diff
@@ -1076,35 +1168,61 @@ void make_fan_data_remove_gaps(FanProjData& fan_data,
 }
 
 
-/// **** This function make proj_data from fan_data while adding the intermodule gaps **** ////
-/// *** fan_data doesn't have gaps, proj_data has gaps *** ///
-void set_fan_data_add_gaps(ProjData& proj_data,
-                  const FanProjData& fan_data, const float gap_value)
+void make_fan_data_remove_gaps(FanProjData& fan_data,
+                               const ProjData& proj_data)
 {
     int num_rings;
     int num_detectors_per_ring;
     int fan_size;
     int max_delta;
-    shared_ptr<const ProjDataInfoCylindricalNoArcCorr> proj_data_info_ptr =
-    get_fan_info(num_rings, num_detectors_per_ring, max_delta, fan_size,
-                 *proj_data.get_proj_data_info_sptr());
+    auto fan_info_sptr = (get_fan_info(num_rings, num_detectors_per_ring, max_delta, fan_size,
+                                       *proj_data.get_proj_data_info_sptr()));
+    const ProjDataInfo& proj_data_info =*fan_info_sptr.get();
```

Comment:
as far as I can see, `*fan_info_sptr` is equal to `*proj_data.get_proj_data_info_sptr()`. I think therefore the following is much clearer
```suggestion
    const ProjDataInfo& proj_data_info =*proj_data.get_proj_data_info_sptr();
    get_fan_info(num_rings, num_detectors_per_ring, max_delta, fan_size, proj_data_info);
```